polaris/hub/client.py

```diff
@@ -302,14 +308,41 @@ def get_dataset(
         Returns:
             A `Dataset` instance, if it exists.
         """
+        return self._get_dataset(owner, name, ArtifactSubtype.STANDARD.value, verify_checksum)
 
+    def _get_dataset(
+        self,
+        owner: Union[str, HubOwner],
+        name: str,
+        artifact_type: ArtifactSubtype,
+        verify_checksum: bool = True,
+    ) -> Dataset:
+        """Loads either a standard or competition dataset from Polaris Hub
+
+        Args:
+            owner: The owner of the dataset. Can be either a user or organization from the Polaris Hub.
+            name: The name of the dataset.
+            artifact_type: indicates whether the artifact is of the standard or competition type.
+            verify_checksum: Whether to use the checksum to verify the integrity of the dataset.
+
+        Returns:
+            A `Dataset` instance, if it exists.
+        """
         with ProgressIndicator(
-            start_msg="Fetching dataset...",
-            success_msg="Fetched dataset.",
+            start_msg="Fetching artifact...",
+            success_msg="Fetched artifact.",
             error_msg="Failed to fetch dataset.",
         ):
-            response = self._base_request_to_hub(url=f"/dataset/{owner}/{name}", method="GET")
-            storage_response = self.get(response["tableContent"]["url"])
+            url = (
+                f"/v1/dataset/{owner}/{name}"
+                if artifact_type == ArtifactSubtype.STANDARD.value
```

Comment:
**Suggestion**
Instead of passing in an enum to denote the concrete type we're looking for, we could pass in the class itself (`Dataset` or `CompetitionDataset`), and set the endpoint URL pattern, dataset info field name, and md5 sum field name, as class attributes.

We have specific classes that identify each artifact type, but we scatter their specific information across the code.

If we want to keep the information related to the client separate from the artifact data structure, I'd suggest putting all these information in an `ArtifactType` enum. Something like this:
```python
class ArtifactType(Enum):
    DATASET = ("dataset", "{api_url}/v1/dataset/{owner}/{name}", Dataset)
    BENCHMARK = ("benchmark", "{api_url}/v1/benchmark/{owner}/{name}", BenchmarkSpecification)
    COMPETITION = ("competition", "{api_url}/v2/competition/{owner}/{name}", Competition)
    COMPETITION_DATASET = ("competition_dataset", "{api_url}/v2/competition/datasets/{owner}/{name}", CompetitionDataset)

    def __init__(self, name: str, url_pattern: str, model: BaseArtifactModel):
        self.name = name
        self.url_pattern = url_pattern
        self.model = model

    def get_url(self, owner: str, name: str):
        return self.url_pattern.format(api_url=settings.api_url, owner=owner, name=name)
```

Reply:
Great idea, I'll write a ticket for this.